typing/ctype.ml

```diff
@@ -2526,16 +2548,12 @@ let constrain_type_jkind ~fixed env ty jkind =
      like this that can occur, though, and may need a more principled solution
      later).
 
-     As this unboxes types, it might unbox an existential type. We thus keep
-     track of whether [ty] [is_open]. EDIT: This is actually pointless and
-     could be removed: #3684.
```

Comment:
Note to self: Mention that letting the variables escape is intentioned, as in richard's pr.